quake1_example/class/func_door.gd

```diff
@@ -4,77 +4,115 @@ class_name QmapbspQuakeFunctionDoor
 # this is written from scratch. No QuakeC code was applied here
 # maybe not accurate to the original approach
 
-var tween : Tween
-var add : Vector3
-var add_reveal : Vector3
-var dura : float
-var wait : int
-var calc_ : bool = false
-var streams : Array
-var viewer : QmapbspQuakeViewer
-var open : bool = false
+# Spawnflags:
+# 1 = Starts Open
+# 4 = Don't link
+# 8 = Gold Key required
+# 16 = Silver Key required
+# 32 = Toggle
+const START_OPEN : int = 1;
+const DONT_LINK : int = 4;
+const GOLD_KEY : int = 8;
+const SILVER_KEY : int = 16;
+const TOGGLE : int = 32;
+# Properties defaults:
+const LIP : int = 8
+const WAIT : int = 3
+const SPEED : int = 100
+const DMG : int = 2
 
-var player : AudioStreamPlayer3D
-var player_end : bool = false
-var links : Array[QmapbspQuakeFunctionDoor]
+const TRIGGER_PADDING : Vector3 = Vector3(1.8, 0.25, 1.8)
 
-signal emit_message_once(m : String)
-
-# according to the QC file
+# 0: "Silent"
+# 1: "Stone"
+# 2: "Machine"
+# 3: "Stone Chain"
+# 4: "Screechy Metal"
 const audio_paths := [
-	['', ''],
+	['misc/null.wav', 'misc/null.wav'],
 	['doors/drclos4.wav', 'doors/doormv1.wav'],
 	['doors/hydro1.wav', 'doors/hydro2.wav'],
 	['doors/stndr1.wav', 'doors/stndr2.wav'],
-	['doors/ddoor1.wav', 'doors/ddoor2.wav'],
-	['', ''],
+	['doors/ddoor1.wav', 'doors/ddoor2.wav']
 ]
 
-var func_door := [
+const keys_audio_paths := [
 	['doors/medtry.wav', 'doors/meduse.wav'],
 	['doors/runetry.wav', 'doors/runeuse.wav'],
 	['doors/basetry.wav', 'doors/baseuse.wav'],
 ]
 
+# 0 : "Medieval"
+# 1 : "Metal (runic)"
+# 2 : "Base"
+const gold_keys_message = [
```

Comment:
If the keys are required to open doors, that means we have to implement key entities. Isn't it ? ^^"

Reply:
Of course :) this just makes the doors behave as if that was the case. We handle keys and such in an inventory for the player, but that's not for this PR IMHO.

Reply:
We will implement keys, and pickups, and everything else at some point :)